src/api/java/com/minecolonies/api/tileentities/AbstractTileEntityGrave.java

```diff
@@ -0,0 +1,149 @@
+package com.minecolonies.api.tileentities;
+
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.IGraveData;
+import net.minecraft.inventory.container.INamedContainerProvider;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.tileentity.ITickableTileEntity;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityType;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import net.minecraftforge.items.ItemStackHandler;
+import org.jetbrains.annotations.Nullable;
+
+import javax.annotation.Nonnull;
+
+import static com.minecolonies.api.research.util.ResearchConstants.RESURRECT_CHANCE;
+import static com.minecolonies.api.util.constant.Constants.*;
+
+public abstract class AbstractTileEntityGrave extends TileEntity implements INamedContainerProvider
+{
+    /**
+     * The inventory of the tileEntity.
+     */
+    protected ItemStackHandler inventory;
```

Comment:
shouldn't we make this a rack type instead then? (instead of copying a lot of this)

Reply:
I'm not sure, I want to avoid this grave to interact with other racks, warehouse, ext. What if a citizen die next to an existing Single rack, if we make this a rack type, the rack could combine in double for instance. They would also be included in many unrelated code. I personally feel it is better to keep the grave an the rack separated.

Reply:
shouldn't be possible, you can force them to be single (like we do with the huts, they are also all AbstractRacks internally)

Reply:
Mmmm, I will look into it and adapt then (y)